src/slack.rs

```diff
@@ -96,12 +97,19 @@ impl Slack {
 
         if !self.is_unique(&slack_msg) {
             info!("Skipping duplicate message to {}", channel);
-            return Ok(());
+            return;
         }
 
         info!("Sending message to #{}", channel);
 
-        self.client.post_void("", &slack_msg)
+        self.client.spawn(self.client.post_void_async("", &slack_msg).then(|res| {
+            match res {
+                Ok(Ok(_)) => info!("Successfully sent slack message"),
+                Ok(Err(e)) => error!("Error sending slack message: {}", e),
+                Err(_) => error!("Unknown error sending slack message"),
+            };
+            future::ok::<(), ()>(())
```

Comment:
All these unit type primitives look strange here, but I get that this is required by the tokio event loop.